app/mailboxes/imap/imap_mailbox.rb

```diff
@@ -54,7 +54,7 @@ def find_message_by_references
 
     return if @inbound_mail.references.blank?
 
-    references = @inbound_mail.references
+    references = Array.wrap(@inbound_mail.references)
```

Comment:
Approving this assuming that `@inbound_mail.references` will always be a string

Reply:
@tejaswinichile is it possible for references to be of any other form, say a string of comma separated IDs?